l2geth/eth/api_tracer.go

```diff
@@ -518,6 +527,137 @@ func (api *PrivateDebugAPI) traceBlock(ctx context.Context, block *types.Block,
 	return results, nil
 }
 
+// TraceCall lets you trace a given eth_call. It collects the structured logs
+// created during the execution of EVM if the given transaction was added on
+// top of the provided block and returns them as a JSON object.
+func (api *PrivateDebugAPI) TraceCall(ctx context.Context, args ethapi.CallArgs, blockNrOrHash rpc.BlockNumberOrHash, config *TraceCallConfig) (interface{}, error) {
+	statedb, header, err := api.eth.APIBackend.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+
+	if statedb == nil || err != nil {
+		return nil, fmt.Errorf("Error retrieving state")
+	}
+
+	// Set sender address or use a default if none specified
+	var addr common.Address
+	if args.From == nil {
+		if !rcfg.UsingBVM {
+			if wallets := api.eth.AccountManager().Wallets(); len(wallets) > 0 {
+				if accounts := wallets[0].Accounts(); len(accounts) > 0 {
+					addr = accounts[0].Address
+				}
+			}
+		}
+	} else {
+		addr = *args.From
+	}
+
+	// Override the fields of specified contracts before execution.
+	if config != nil {
+		for addr, account := range *config.StateOverrides {
+			// Override account nonce.
+			if account.Nonce != nil {
+				statedb.SetNonce(addr, uint64(*account.Nonce))
+			}
+			// Override account(contract) code.
+			if account.Code != nil {
+				statedb.SetCode(addr, *account.Code)
+			}
+			// Override account balance.
+			if account.Balance != nil {
+				statedb.SetBalance(addr, (*big.Int)(*account.Balance))
+			}
+			if account.State != nil && account.StateDiff != nil {
+				return nil, fmt.Errorf("account %s has both 'state' and 'stateDiff'", addr.Hex())
+			}
+			// Replace entire state if caller requires.
+			if account.State != nil {
+				statedb.SetStorage(addr, *account.State)
+			}
+			// Apply state diff into specified accounts.
+			if account.StateDiff != nil {
+				for key, value := range *account.StateDiff {
+					statedb.SetState(addr, key, value)
+				}
+			}
+		}
+	}
+
+	// Set default gas & gas price if none were set
+	gas := uint64(math.MaxUint64 / 2)
+	if args.Gas != nil {
+		gas = uint64(*args.Gas)
+	}
+
+	// SuggestGasPrice implements ContractTransactor.SuggestGasPrice. Since the simulated
+	// chain doesn't have miners, we just return a gas price of 1 for any call unless specified differently
+	gasPrice := big.NewInt(1)
+	if args.GasPrice != nil {
+		gasPrice = args.GasPrice.ToInt()
+	}
+
+	value := new(big.Int)
+	if args.Value != nil {
+		value = args.Value.ToInt()
+	}
+
+	var data []byte
+	if args.Data != nil {
+		data = []byte(*args.Data)
+	} else {
+		return nil, fmt.Errorf("Data field is empty")
```

Comment:
If a transaction is a native token transfer transaction, then the error will be returned. Is this intentional?

Reply:
Good point, let me fix that